### PR TITLE
Add a workflow job to automate uploading to testpypi

### PR DESCRIPTION
**Pull request by ngoldbaum**
Not sure how to make ASCII art matching the other jobs

### CODE REVIEW

**Comment by timkpaine** [comment]
`7` is used for adapter tests but grab `8` https://github.com/python-project-templates/cpp/blob/e09da5c92ba382218a558b333975397ffcf901de/.github/workflows/build.yml#L444

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -616,3 +616,33 @@ jobs:
...
+      runs-on: ubuntu-22.04
+      permissions:
+        id-token: write
+      steps:
+        - name: Download wheels and sdist
+          uses: actions/download-artifact@v4
+          with:
+            name:
+            merge-multiple: true
+            path: ~/dist
```
[the remark this answers is not in the document]
Comment (ngoldbaum) [reply .github/workflows/build.yml:638]: The action expect it to be in a folder named `dist`, changed here and below to `./dist`.

**Comment by ngoldbaum** [comment]
The other issue is that right now the version is static so testpypi won't let us upload a new release with every merged PR. I think `scikit-build` can work with `setuptools_scm` - did you ever look at that or any other kind of dynamic version generation from VCS versions?

**Comment by ngoldbaum** [comment]
For now just set it to only upload if a tag is added.

**Comment by timkpaine** [comment]
Can you make a tag like `deleteme` and push to your fork, and show the successful run just to verify

**Comment by ngoldbaum** [comment]
Thanks for the suggestion to try on my fork. The way I originally configured it there were some bugs. Also I got tripped up on the check for tags that start with `v` so `deleteme` doesn't work but `vdeleteme` does.

Here's the action run https://github.com/ngoldbaum/csp/actions/runs/7879281136/job/21501011776

It failed to upload because my fork isn't a trusted publisher to testpypi for the `csp` pacakge. Would you like me to temporarily add it to make sure the upload goes through? We can also merge this as-is and debug once we push an e.g. `v0.1.0` tag to the main repo.

**Review by trallard — APPROVED** [review]
This looks good to me so I will go ahead and approve @ngoldbaum.
That way we can test the release and iterate as needed

**Review by trallard — DISMISSED** [review]

**Review by timkpaine — APPROVED** [review]

**Comment by ngoldbaum** [comment]
It's not clear to me why github is saying I'm dismissing stale reviews, it seems to be happening automatically.

**Review by timkpaine — APPROVED** [review]

**Pull request merged**